### PR TITLE
Changelog and development model updates

### PR DESCRIPTION
**Pull request by brycelelbach**

### CODE REVIEW

File: doc/changelog.md [hunk 1]
```diff
@@ -1,30 +1,201 @@
...
+  Thanks to Rong Ou for this contribution.
+- #1131: Fix the one-policy overload of `thrust::async::copy` to not copy the
+    policy, resolving use-afer-move issues.
+- #1145: When cleaning up type names in `unittest::base_class_name`, only call
+    `std::string::replace` if we found the substring we are looking to replace.
+- #1139: Don't use `cxx::__demangle` in NVC++.
+- #1102: Don't use `thrust::detail::normal_distribution_nvcc` for Feta because
+    it uses `erfcinv`, a non-standard function that Feta doesn't have.
+
+# Thrust 1.9.9 (CUDA Toolkit 11.0 Early Access)
```
Comment (dkolsen-pgi) [inline doc/changelog.md:50]: Should this be CUDA 11.0 GA rather than EA?

File: doc/development_model.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+The version number for a Thrust release uses the following format:
+`MMM.mmm.ss-ppp`, where:
+
+   * `THRUST_VERSION_MAJOR`/`MMM`: Major version, up to 3 decimal digits. It is incremented
+     when the fundamental nature of the library evolves, leading to widespread changes across the
+     entire library interface with no guarantee of API, ABI, or semantic compatibility with former
+     versions.
+   * `THRUST_VERISON_MINOR`/`mmm`: Minor version, up to 3 decimal digits. It is incremented when
+     breaking API, ABI, or semantic changes with are made.
```
Comment (dkolsen-pgi) [inline doc/development_model.md:44]: The grammar of this sentence is wrong, "or semantic changes with are made".  I don't know what was intended.

File: doc/changelog.md (hunk 1 above)
Comment (brycelelbach) [reply doc/changelog.md:50]: Uh yes, yes it should.

File: doc/development_model.md (hunk 2 above)
Comment (brycelelbach) [reply doc/development_model.md:44]: Yes those are bad words they will be punished.

File: doc/changelog.md [hunk 3]
```diff
@@ -1,30 +1,208 @@
...
+
+## New Features
+
+- #1130: CMake `find_package` support.
+  This is significant because there is a legacy `FindThrust.cmake` script
+    authored by a third party in widespread use in the community which has a
+    bug in how it parses Thrust version numbers which will cause it to
+    incorrectly parse 1.9.10.
+  This script only parses the first digit of each part of the Thrust version
+    number: for example, Thrust 17.17.17 would be interpreted as Thrust 1.1.1.
```
Comment (griwes) [inline doc/changelog.md:18]: This is not accurate; it also assumed fixed number of `0`s between the significant digits. We should probably just say that it assumed single digit version components and broke down when that was not the case.

File: doc/changelog.md [hunk 4]
```diff
@@ -1,30 +1,208 @@
...
+
+# Thrust 1.9.8-1 (NVIDIA HPC SDK 20.3)
+
+## Summary
+
+Thrust 1.9.8-1 is a variant of 1.9.8 accompanying the NVIDIA HPC SDK 20.3.
+It contains modifications necessary to serve as the implementation of NVC++'s
+  GPU-accelerated C++17 Parallel Algorithms.
+
+# Thrust 1.9.8 (CUDA Toolkit 11.0)
```
Comment (griwes) [inline doc/changelog.md:145]: And this should be EA.

File: doc/changelog.md [hunk 5]
```diff
@@ -1,30 +1,208 @@
...
+
+Thrust 1.9.8, which is included in the CUDA Toolkit 11.0 release, removes
+  Thrust's internal derivative of CUB, upstreams all relevant changes too CUB,
+  and adds CUB as a Git submodule.
 It will now be necessary to do `git clone --recursive` when checking out
   Thrust, and to update the CUB submodule when pulling in new Thrust changes.
 Additionally, CUB is now included as a first class citizen in the CUDA toolkit.
-Thrust v1.9.8 also fixes bugs preventing most Thrust algorithms from working
+Thrust 1.9.8 also fixes bugs preventing most Thrust algorithms from working
   with more than `2^32` elements.
```
Comment (griwes) [inline doc/changelog.md:156]: Uh... I know this is old, but this should say `2^31-1`, not `2^32`. Nitpick. :P

File: doc/changelog.md (hunk 3 above)
Comment (brycelelbach) [reply doc/changelog.md:18]: I'd still like to have an example. Can we just think up an accurate one?

File: doc/changelog.md (hunk 5 above)
Comment (brycelelbach) [reply doc/changelog.md:156]: Fixed.

File: doc/changelog.md (hunk 3 above)
Comment (griwes) [reply doc/changelog.md:18]: I think 17.17.17 would make it actually not match and create a malformed version string. Should be easy to test.

Comment (griwes) [reply doc/changelog.md:18]: I changed the version int to `101009` locally, got this:
```
-- Found Thrust: /usr/local/cuda/include (found version "1.1.09") 
```

Changed it to `1701717`, which I believe is the correct version literal for 17.17.17, got this:
```
-- Found Thrust: /usr/local/cuda/include (found version "1.1.1701717") 
```

For `1000100` (so 10.1.0):
```
-- Found Thrust: /usr/local/cuda/include (found version "1.0.0100") 
```

**Pull request merged**